src/e2e/src/pages/blotter.component.ts

```diff
@@ -1,70 +1,98 @@
-        tradeDealtCCY: root.element(
-          by.css(
-            '[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="dealtCurrency"]'
-          )
-        ),
-        tradeNotional: root.element(
-          by.css(
-            '[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="notional"]'
-          )
-        ),
-        tradeRate: root.element(
-          by.css(
-            '[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="spotRate"]'
-          )
-        ),
-        tradeValueDate: root.element(
-          by.css(
-            '[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="valueDate"]'
-          )
-        ),
-        tradeBackGroundColour: root.element(
-          by.css(
-            '[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="statusIndicator"]'
-          )
-        )
-      }
-    }
+    // this.tradesTable = {
+    //   executedTrades: {
+    //     latestTrade: root.element(by.css("[col-id='tradeId']" + ".ag-cell")),
+    //     tradeID: root.element(by.qa(`306-tradeId`)),
+    //     tradeStatus: root.element(by.qa(`1-status`)),
+    //     tradeDate: root.element(by.qa(`1-tradedate`)),
+    //     tradeDirection: root.element(by.qa(`1-direction`)),
+    //     tradeCCYCCY: root.element(by.qa(`1-symbol`)),
+    //     tradeDealtCCY: root.element(by.qa(`1-dealtCurrency`)),
+    //     tradeNotional: root.element(by.qa(`1-notional`)),
+    //     tradeRate: root.element(by.qa(`1-spotRate`)),
+    //     tradeValueDate: root.element(by.qa(`1-valueDate`)),
+    //     tradeBackGroundColour: root.element(
+    //       by.css(
+    //         '[data-qa="shell-route__blotter-wrapper"] .ag-body .ag-row:nth-child(1) [col-id="statusIndicator"]'
+    //       )
+    //     )
+    //   }
+    // }
+  }
+  async getTradeId(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-tradeId`))
+    return trade
+  }
+
+  async getTradeStatus(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-status`))
+    return trade
+  }
+
+  async getTradeDate(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-tradeDate`))
+    return trade
+  }
+
+  async getTradeDirection(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-direction`))
+    return trade
+  }
+
+  async getTradeCCYCCY(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-symbol`))
+    return trade
+  }
+
+  async getTradeDealtCCY(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-dealtCurrency`))
+    return trade
+  }
+
+  async getTradeNotional(num: string): Promise<ElementFinder> {
+    const trade = await this.root.element(by.qa(`${num}-notional`))
+    return trade
```

Comment:
To make this even cleaner, you could just return the await without the `const` assignment.

`return await this.root.element(by.qa(`${num}-notional`))`

Same can be applied to the other functions